File: CS/NavigationFrameWizard/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NavigationFrameWizard {
    public class Order : INotifyPropertyChanged {

        protected DateTime _Date;

        public DateTime Date {
            get {
                return this._Date;
            }

            set {
                if(this._Date != value) {
                    this._Date = value;
                    this.OnPropertyChanged("Date");
                }
            }
        }
        
        protected string _Description;

        public string Description {
            get {
                return this._Description;
            }

            set {
                if(this._Description != value) {
                    this._Description = value;
                    this.OnPropertyChanged("Description");
                }
            }
        }

        public void OnPropertyChanged(string info) {
            if(this.PropertyChanged != null) {
                this.PropertyChanged(this, new PropertyChangedEventArgs(info));
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
    }
}

File: VB/NavigationFrameWizard/Order.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace NavigationFrameWizard
    Public Class Order
        Implements INotifyPropertyChanged

        Protected _Date As Date

        Public Property [Date]() As Date
            Get
                Return Me._Date
            End Get

            Set(ByVal value As Date)
                If Me._Date <> value Then
                    Me._Date = value
                    Me.OnPropertyChanged("Date")
                End If
            End Set
        End Property

        Protected _Description As String

        Public Property Description() As String
            Get
                Return Me._Description
            End Get

            Set(ByVal value As String)
                If Me._Description <> value Then
                    Me._Description = value
                    Me.OnPropertyChanged("Description")
                End If
            End Set
        End Property

        Public Sub OnPropertyChanged(ByVal info As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(info))
        End Sub

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
    End Class
End Namespace
